pkg/controller/apiclient/apiclient_util_test.go

```diff
@@ -14,18 +14,180 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package apiclient
+package apiclient_test
```

Comment:
Humbly, I think both views on how to resolve the would-be dependency cycle here are wrong. If you look at these tests, they are very specifically testing the in-memory implementation of the APIClient interface. That means these tests belong alongside that implementation _in_ the `pkg/controller/apiclient/mem` package.

Also, although some tests similar to this do already exist and aren't completely useless, more tests of this ilk might not give us the biggest bang for the buck. The in-memory implementation of this interface is, essentially, a fake implementation used only to facilitate tests. These are tests for test code. As I said-- that's not completely without value, but as @pmorie has indicated, longer term we want to stop using this implementation and use the reactor chain pattern here instead. (Which will courtesy of some code generation that I don't claim to understand.)

Bottom line-- I would suggest that we move this to where it belongs and not fuss over it too much further since its value is limited.